cmake/svs.cmake

```
@@ -0,0 +1,84 @@
+cmake_minimum_required(VERSION 3.12)
+cmake_policy(SET CMP0074 NEW)
+if(POLICY CMP0135)
+    cmake_policy(SET CMP0135 NEW)
+endif()
+
+# Valgrind does not support AVX512 and Valgrind in running in Debug
+# so disable it if we are in Debug mode
+string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
+if(uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+    set(SVS_NO_AVX512 ON)
+endif()
+
+if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "(x86_64)|(AMD64|amd64)")
+    set(SVS_SUPPORTED 1)
+else()
+    set(SVS_SUPPORTED 0)
+    message(STATUS "SVS is not supported on this architecture")
+endif()
+
+# GCC < v11 does not support C++20 features required for SVS
+# https://gcc.gnu.org/projects/cxx-status.html
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.0")
+        set(SVS_SUPPORTED 0)
+        message(STATUS "Insufficient gcc version for SVS")
```

please specify the minimal version required
```suggestion
        message(STATUS "Skipping SVS: requires GCC >= 11.")
```

---

Done.